common/src/main/java/com/wynntils/features/user/redirects/BlacksmithRedirectFeature.java

```diff
@@ -0,0 +1,112 @@
+public class BlacksmithRedirectFeature extends UserFeature {
+    private static final Pattern BLACKSMITH_MESSAGE_PATTERN = Pattern.compile(
+            "§5Blacksmith: §r§dYou (.+): (.+) for a total of §r§e(\\d+)§r§d (emeralds|scrap). It was a pleasure doing business with you.");
+    private static final Pattern ITEM_PATTERN = Pattern.compile("§r§([fedacb53])(.+?)§r§d");
+
+    @Override
+    public List<Class<? extends Model>> getModelDependencies() {
+        return List.of(ChatModel.class);
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onChat(ChatMessageReceivedEvent event) {
+        Matcher messageMatcher = BLACKSMITH_MESSAGE_PATTERN.matcher(event.getCodedMessage());
+        if (!messageMatcher.matches()) return;
+        event.setCanceled(true);
+
+        EnumMap<ItemTier, Integer> totalItems = new EnumMap<>(ItemTier.class);
+        // How many emeralds/scrap we got from the transaction.
+        String paymentString = messageMatcher.group(3);
+        // Tracks count of sold or scrapped items
+        int totalItemInteger = 0;
+        // Full message to send to the user.
+        String sendableMessage = "";
+
+        // This is for selling items for emeralds.
+        if (messageMatcher.group(1).equals("sold me")) {
+
+            // Retrieve the color code of the item, and then match it to the item tier.
+            Matcher itemMatcher =
+                    ITEM_PATTERN.matcher(messageMatcher.group(2)); // Second group contains all of the items.
+            // Tally up the items that we sold.
+            while (itemMatcher.find()) {
+                ChatFormatting itemColor = ChatFormatting.getByCode(itemMatcher
+                        .group(1)
+                        .charAt(0)); // find the color code to find ChatFormatting and ascertain the tier.
+                ItemTier tierToIncrease = ItemTier.fromChatFormatting(itemColor);
+                if (tierToIncrease == null) continue;
+                totalItems.put(tierToIncrease, totalItems.getOrDefault(tierToIncrease, 0) + 1);
+                totalItemInteger++;
+            }
+            String itemPluralizer = totalItemInteger == 1 ? "item" : "items";
+
+            // Build up the string that outlines how many items were sold in what tier (0/0/0/0/0/0/0/0).
+            StringBuilder countByTier = new StringBuilder();
+            for (ItemTier tier : ItemTier.values()) {
+                countByTier.append('/' + tier.getChatFormatting().toString() + totalItems.getOrDefault(tier, 0));
+                countByTier.append(ChatFormatting.LIGHT_PURPLE);
+            }
+            countByTier.append(")");
+            countByTier.setCharAt(0, '(');
+
+            String countByTierString = countByTier.toString();
+
+            // Sold 1 (1/0/0/0/0/0/0/0) item for 4e.
+            sendableMessage = String.format(
+                    "%sSold %d %s %s for %s%s%s%s.",
+                    ChatFormatting.LIGHT_PURPLE,
```

Comment:
You can include colors int the format string itself to make this look less weird